### PR TITLE
Make log ERROR->INFO

### PR DESCRIPTION
**Pull request by hsaraogi**
This changes the log level and log message for #799. 

The ``ignoreHotspottingChecks`` flag is just being used other than to decide if we should print this info. 

Should we deprecate ``ignoreHotspottingChecks()`` and print the info everytime? @rhero.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/947)

### CODE REVIEW

**Comment by clockfort** [comment]
? Who is going to actually read these if they always get spewed out all the time no matter what at INFO level ?
If you want to do this you might as well remove the feature entirely

**Comment by rhero** [comment]
I think the issue we had with ERROR is that the person reading them can't really do anything about it, so in that sense, I'd be okay with just removing the logging too. I just think we're doing more harm than good by keeping it at ERROR.

**Comment by hsaraogi** [comment]
I think the log message(now info) should be kept around for people doing range scans and aware of the behavior, just to signal hot-spotting. Since the log is just info, I thought it would be fine showing to everyone, but on second thoughts ``ignoreHotspottingChecks`` probably means that they are aware of potential hotspots and are confident they'll not encounter them.

**Comment by jboreiko** [comment]
The ideal state of the world here is that all products take the time to resolve these errors so that no end user is forced to deal with them.  I have attached the call hierarchy which makes it clear why it shows up at startup (create/delete/truncate) as well as at the time we want (when building the project in *.generateSchema()).
screen shot 2016-09-16 at 3 52 33 pm

Running the build of an internal product I clearly see the errors reported:

```:<redacted>:manifestClasspathJar SKIPPED
15:53:34.097 [main] ERROR c.p.a.t.d.AbstractDefinition - First row component bucket of type VAR_STRING will likely cause hot-spotting with the partitioner in Cassandra. This is caused by the structure of variable-sized types which will state their length prior to the value resulting in them being partition predominantly by the LENGTH of the values which is likely to be similar. If you anticipate never running on Cassandra or feel you can safely ignore this case (for instance, if this table will never be very large) then this error can be safely ignored by adding ignoreHotspottingChecks() to the table schema. In a future release atlas will fail to start if there are hotspotting issues that have not been ignored. (This error is directed at the developer of this atlas application, who should be informed that they need to change their schema)
15:53:34.101 [main] ERROR c.p.a.t.d.AbstractDefinition - First row component bucket of type VAR_STRING will likely cause hot-spotting with the partitioner in Cassandra. This is caused by the structure of variable-sized types which will state their length prior to the value resulting in them being partition predominantly by the LENGTH of the values which is likely to be similar. If you anticipate never running on Cassandra or feel you can safely ignore this case (for instance, if this table will never be very large) then this error can be safely ignored by adding ignoreHotspottingChecks() to the table schema. In a future release atlas will fail to start if there are hotspotting issues that have not been ignored. (This error is directed at the developer of this atlas application, who should be informed that they need to change their schema)
15:53:34.102 [main] ERROR c.p.a.t.d.AbstractDefinition - First row component eventId of type VAR_LONG will likely cause hot-spotting with the partitioner in Cassandra. This is caused by the structure of variable-sized types which will state their length prior to the value resulting in them being partition predominantly by the LENGTH of the values which is likely to be similar. If you anticipate never running on Cassandra or feel you can safely ignore this case (for instance, if this table will never be very large) then this error can be safely ignored by adding ignoreHotspottingChecks() to the table schema. In a future release atlas will fail to start if there are hotspotting issues that have not been ignored. (This error is directed at the developer of this atlas application, who should be informed that they need to change their schema)
:<redacted>-atlas:compileGeneratedJava UP-TO-DATE```

I would be fine removing it from the startup code paths but leave it on the generate one.  But this really signals that we should be reaching out to products with this problem to ensure they resolve it and leave with a better understanding of the trade offs being made in their schemas.

**Comment by schlosna** [comment]
Can't we make the build time schema generation return a non-zero response and fail compilation so that a dev has to explicitly update the schema to ignore the non-ideal partitioning?

**Comment by jboreiko** [comment]
I like this idea, though we would have to be ready to help people otherwise they might just slap a bunch of the ignoreChecks on everything to be able to move forward.  I.e. lets go around an preempt the break by asking teams to fix this and then we can break it.

**Comment by hsaraogi** [comment]
Closing in favor of displaying errors until we enforce check on schema generation.

**Comment by rhero** [comment]
Did not see that we just closed this out without addressing it. Not in sync that we should be keep the ERROR logs at runtime while waiting for a better solution. I think logging INFO (and not deprecating ignoreHotspottingChecks) or logging nothing are better than currently logging ERROR.

**Comment by jboreiko** [comment]
This will now become a breaking change for all products that have yet to respond to this issue.

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @Test
+    public void testNoFailureWhenHotspottingIgnored() {
+        getIgnoredHotspottingSchema().validate();
+    }
+
+    @Test
+    public void testSuccessfulGenerationWhenHotspottingIgnored() throws IOException {
+        TemporaryFolder temporaryFolder = new TemporaryFolder();
```
Comment (schlosna) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: do you want the `temporaryFolder` to be an instance `@Rule` to get cleaned up?

**Comment by clockfort** [comment]
lgtm but definitely needs release notes because its a blocking change for people

**Comment by gsheasby** [comment]
Release notes added, should be ready to merge once the build is green.

**Pull request merged**